File: Day 7/Longest Repeating Character Replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = defaultdict(int)
        left,max_freq,best = 0,0,0

        for right,ch in enumerate(s):
            count[ch] +=1
            max_freq = max(max_freq, count[ch])
            while (right - left +1) - max_freq > k:
                count[s[left]] -= 1
                left += 1
           
            best = max(best, right - left +1)
       
        return best

File: Day 7/Longest Repeating Character Replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> freq(26, 0);
        int left = 0, best = 0, maxFreq = 0;

        for (int right = 0; right < (int)s.size(); ++right) {
            int r = s[right] - 'A';
            freq[r]++;
            maxFreq = max(maxFreq, freq[r]);

            while ((right - left + 1) - maxFreq > k) {
                freq[s[left] - 'A']--;
                left++;
            }

            best = max(best, right - left + 1);
        }
        return best;
    }
};
